pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -210,11 +210,19 @@ func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
 		} else if newSvcAnnotation == "true" {
 			podKeys = sets.NewString(c.getPodsFromService(newSvc)...)
 		}
-	} else if !reflect.DeepEqual(oldSvc.Spec.Selector, newSvc.Spec.Selector) {
+	}
+
+	var oldPodSet, newPodSet sets.String
+	if oldSvc.Spec.Type != newSvc.Spec.Type {
+		newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
+		podKeys = podKeys.Union(newPodSet)
+	}
+
+	if !reflect.DeepEqual(oldSvc.Spec.Selector, newSvc.Spec.Selector) {
 		// Disjunctive union of Pods from both Service sets.
-		oldPodSet := sets.NewString(c.getPodsFromService(oldSvc)...)
-		newPodSet := sets.NewString(c.getPodsFromService(newSvc)...)
-		podKeys = oldPodSet.Difference(newPodSet).Union(newPodSet.Difference(oldPodSet))
+		oldPodSet = sets.NewString(c.getPodsFromService(oldSvc)...)
+		newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
+		podKeys = podKeys.Union(oldPodSet.Difference(newPodSet).Union(newPodSet.Difference(oldPodSet)))
```

Comment:
It seems that this code does not really need to execute in case of an annotation change (`oldSvcAnnotation != newSvcAnnotation`), should it be in an `else` block?

If I look at this edge case:
* `oldSvcAnnotation` is true
* `newSvcAnnotation` is false
* `oldSvc.Spec.Type != newSvc.Spec.Type`
* `oldSvc.Spec.Selector != newSvc.Spec.Selector`
You actually end up enqueuing both old Pods and new Pods, when you only need to enqueue old Pods.